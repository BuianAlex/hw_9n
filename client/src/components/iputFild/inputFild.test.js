import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import InputFild from './inputFild';

describe('<InputFild /> tests', () => {
  it('should render as default button', () => {
    const options = {
      type: 'text',
      id: 'login',
      label: 'Login:',
      isRequired: true,
      value: '',
      disabled: false
    };

    const { container } = render(
      <InputFild options={options} onValid={() => {}} />
    );
    const inp = expect(container.firstChild).toMatchSnapshot();
    // console.log(container.firstChild);
  });

  // it('button should has class', () => {
  //   const color = 'red'

  //   const { container } = render(<Button color={`${color}`} callback={() => {}} />)
  //   expect(container.firstChild).toHaveClass(`button-${color}`)
  // })

  // it('button should has children', () => {
  //   const color = 'red'
  //   const children = 'test'

  //   const { getByText } = render(<Button color={`${color}`} callback={() => {}}>{children}</Button>)
  //   getByText(children)
  // })

  // it('button should call callback on click', () => {
  //   const color = 'red'
  //   const children = 'test'
  //   const content = 'button'
  //   const onClick = jest.fn()

  //   const { getByTestId } = render(<Button color={`${color}`} onClick={onClick}>{children}</Button>)
  //   fireEvent.click(getByTestId(content))
  //   expect(onClick).toHaveBeenCalledTimes(1)
  // })
});
