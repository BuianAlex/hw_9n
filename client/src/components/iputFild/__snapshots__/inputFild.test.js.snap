// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputFild /> tests should render as default button 1`] = `
<div
  class="mui-textfield"
>
  <input
    id="login"
    type="text"
    value=""
  />
  <label
    for="login"
  >
    Login:
  </label>
  <span
    class="errors"
  />
</div>
`;
